static/js/typeahead_helper.js

```diff
@@ -207,17 +207,27 @@ exports.compare_by_sub_count = function (stream_a, stream_b) {
     return stream_a.subscribers.num_items() < stream_b.subscribers.num_items();
 };
 
+// Find which stream comes first in the sidebar listing order
+exports.compare_by_sidebar = function (stream_a, stream_b) {
+    var a_score = stream_data.is_active(stream_a.name) + stream_a.pin_to_top * 2;
+    var b_score = stream_data.is_active(stream_b.name) + stream_b.pin_to_top * 2;
```

Comment:
yep.  The bulk isn't a material cost -- in a large project, code clarity is far more important than brevity.

I'd use `stream_score += 2`, though, BTW; it's a little safer against someone adding a new rule at the top of the group :)